packages/react-reconciler/src/ReactFiber.js

```diff
@@ -498,14 +510,27 @@ export function createFiberFromElement(
   }
 
   fiber = createFiber(fiberTag, pendingProps, key, mode);
-  fiber.type = type;
+  fiber.elementType = type;
+  fiber.type = resolvedType;
   fiber.expirationTime = expirationTime;
 
+  return fiber;
+}
+
+export function createFiberFromElement(
+  element: ReactElement,
+  mode: TypeOfMode,
+  expirationTime: ExpirationTime,
+): Fiber {
+  const fiber = createFiberFromElementWithoutDebugInfo(
```

Comment:
Why the two functions? Doesn't look like this is called anywhere else

Reply:
So that I can do early returns in the other one. Compiler can restructure and get rid of it in prod.

Reply:
I believe you